spec/unit/pops/loaders/loaders_spec.rb

```diff
@@ -234,15 +234,15 @@ def compile_and_get_notifications(code)
     end
 
     it "an autoloaded type can reference an autoloaded type alias from another module when dependency is present in metadata.json" do
-      File.stubs(:read).with(user_metadata_path).returns user_metadata.merge('dependencies' => [ { 'name' => 'test-usee'} ]).to_pson
+      Puppet::FileSystem.stubs(:read).with(user_metadata_path, {:encoding => 'utf-8'}).returns user_metadata.merge('dependencies' => [ { 'name' => 'test-usee'} ]).to_pson
       expect(eval_and_collect_notices(<<-CODE, node)).to eq(['ok'])
         assert_type(User::WithUseeZero, [0])
         notice(ok)
       CODE
     end
 
     it "an autoloaded type can reference an autoloaded type alias from another module when other module has it declared in init.pp" do
```

Comment:
It is unit testing that it has the ability to reference things elsewhere that is should have visibility into. An integration test would require laying out the involved modules, the type aliases in separate files etc. and use no mocking. The reason it is using a function call, and a type is to make the test compact and readable - it would be quite long to do exactly the same in ruby. The same style is used in many places throughout our code base - i.e. using a compilation of puppet code to trigger the exact sequence of calls required to unit test the thing being tested.